test/unit/mem/buf/src/BufTest.cpp

```diff
@@ -162,27 +162,35 @@ TEMPLATE_LIST_TEST_CASE("memBufConstTest", "[memBuf]", alpaka::test::TestAccs)
 template<typename TAcc>
 static auto testAsyncBufferImmutable(alpaka::Vec<alpaka::Dim<TAcc>, alpaka::Idx<TAcc>> const& extent) -> void
 {
-    using Dev = alpaka::Dev<TAcc>;
-    using Pltf = alpaka::Pltf<Dev>;
-    using Queue = alpaka::test::DefaultQueue<Dev>;
-
-    using Elem = float;
-    using Dim = alpaka::Dim<TAcc>;
-    using Idx = alpaka::Idx<TAcc>;
-
-    Dev const dev = alpaka::getDevByIdx<Pltf>(0u);
-    Queue queue(dev);
-
-    // memory is allocated when the queue reaches this point
-    auto const buf = alpaka::allocAsyncBuf<Elem, Idx>(queue, extent);
-
-    // synchronous operations must wait for the memory to be available
-    alpaka::wait(queue);
-    auto const offset = alpaka::Vec<Dim, Idx>::zeros();
-    alpaka::test::testViewImmutable<Elem>(buf, dev, extent, offset);
+    {
+        using Dev = alpaka::Dev<TAcc>;
+        using Pltf = alpaka::Pltf<Dev>;
+        using Queue = alpaka::test::DefaultQueue<Dev>;
+
+        using Elem = float;
+        using Dim = alpaka::Dim<TAcc>;
+        using Idx = alpaka::Idx<TAcc>;
+
+        Dev const dev = alpaka::getDevByIdx<Pltf>(0u);
+        Queue queue(dev);
+
+        // memory is allocated when the queue reaches this point
+        auto const buf = alpaka::allocAsyncBuf<Elem, Idx>(queue, extent);
+
+        // synchronous operations must wait for the memory to be available
+        alpaka::wait(queue);
+        auto const offset = alpaka::Vec<Dim, Idx>::zeros();
+        alpaka::test::testViewImmutable<Elem>(buf, dev, extent, offset);
+
+        // the buffer will queue the deallocation of the memory when it goes out of scope,
+        // and extend the lifetime of the queue until all memory operations have completed.
+        // delay the end end of the queue to push the buffer deletetion task
+        // after all local refs to queue have been dropped
+        alpaka::enqueue(queue, []() { std::this_thread::sleep_for(std::chrono::microseconds(1000)); });
+        [](auto) {}(std::move(queue));
```

Comment:
What is the usefulness of this lambda ? It is executed immediately.

Reply:
The lambda destroys the `queue` handle in the local scope, by moving it into another scope, where it immediately goes out of scope. This there to remove the local `queue` handle before the buffer is destructed (opposite to the normal destruction order), to leave the last handle inside the buffer or its deleter.

Reply:
I see.

But why do you need a lambda ?
Wouldn't moving from the queue in a nested scope be sufficient ?
```suggestion
        {
          // ...
          auto unused = std::move(queue) };
        }
```

Reply:
Might do (apart from the `}` there, but would this not raise an "unused" warning?

Reply:
I would argue that it is equally unused in both cases... 
But if it causes a warning, I guess
```suggestion
        {
          auto unused [[maybe_unused]] = std::move(queue);
        }
```
should do :-/

Reply:
But in case of the lambda the missing parameter name tells the compiler, that it is expected to be unused.